grammar.js

```diff
@@ -17,6 +17,20 @@ module.exports = grammar(C, {
             ...original.members.filter((member) => member.content?.name != '_old_style_function_definition'),
         ),
 
+        _top_level_statement: (_, original) => choice(
```

Comment:
yeah, this was the exact problem that got a similar change failing in HLSL. Should we really allow discard as a top-level statement? It actually only makes sense within a shader.